deepchecks/datasets/regression/avocado.py

```diff
@@ -0,0 +1,131 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""The avocado dataset contains historical data on avocado prices and sales volume in multiple US markets."""
+import typing as t
+import pandas as pd
+import joblib
+from urllib.request import urlopen
+from deepchecks import Dataset
+
+__all__ = ['load_data', 'load_fitted_model']
+
+_MODEL_URL = 'https://figshare.com/ndownloader/files/32393723'
+_FULL_DATA_URL = 'https://figshare.com/ndownloader/files/32393729'
+_TRAIN_DATA_URL = 'https://figshare.com/ndownloader/files/32393732'
+_TEST_DATA_URL = 'https://figshare.com/ndownloader/files/32393726'
+_target = 'AveragePrice'
+_CAT_FEATURES = ['region', 'type']
+
+
+def load_data(data_format: str = 'Dataset', as_train_test: bool = True) -> \
+        t.Union[t.Tuple, t.Union[Dataset, pd.DataFrame]]:
+    """Load and returns the Avocado dataset (regression).
+
+    The avocado dataset contains historical data on avocado prices and sales volume in multiple US markets
+    https://www.kaggle.com/neuromusic/avocado-prices.
+
+    The typical ML task in this dataset is to build a model that predicts the average price of Avocados.
+
+    Dataset Shape:
+        .. list-table:: Dataset Shape
+           :widths: 50 50
+           :header-rows: 1
+
+           * - Property
+             - Value
+           * - Samples Total
+             - 18.2K
+           * - Dimensionality
+             - 14
+           * - Features
+             - real, string
+           * - Targets
+             - real 0.44 - 3.25
+
+    Description:
+        .. list-table:: Dataset Description
+           :widths: 50 50
+           :header-rows: 1
+
+           * - Column name
```

Comment:
Had a look at kaggle's description for this dataset...
I think we should add also the description of the columns that have one:
![image](https://user-images.githubusercontent.com/33841818/147591062-ac13617a-c009-4b1c-b8e7-934d9a851490.png)
(including an inline explanation for "what is PLU")

Reply:
![image](https://user-images.githubusercontent.com/20860465/147639800-a581db77-708c-43f1-8d08-70522170a558.png)